operator/pkg/controller/clusterconfig/virtualippool.go

```diff
@@ -0,0 +1,198 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+	metalk8sscalitycomv1alpha1 "github.com/scality/metalk8s/operator/api/v1alpha1"
+	"github.com/scality/metalk8s/operator/pkg/common"
+	"github.com/scality/metalk8s/operator/pkg/controller/utils"
+)
+
+// Name of the namespace used for Virtual IPs
+const vipNamespaceName = "metalk8s-vips"
+
+func (r *ClusterConfigReconciler) reconcileVirtualIPPools(ctx context.Context, reqLogger logr.Logger) utils.SubReconcilerResult {
+	logger := reqLogger.WithValues("SubReconciler", "VirtualIPPool")
+	logger.Info("reconciling VIPs: START")
+	defer logger.Info("reconciling VIPs: STOP")
+
+	pools := r.instance.Spec.WorkloadPlane.VirtualIPPools
+
+	handler := utils.NewObjectHandler(&r.instance, r.client, r.scheme, logger, componentName, appName)
+
+	// Reconcile the VIP namespace
+	namespaceInstance := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: vipNamespaceName}}
+
+	if len(pools) == 0 {
+		// The namespace shouldn't exists
+		changed, err := handler.CreateOrUpdateOrDelete(ctx, nil, []client.Object{namespaceInstance}, nil)
+		if err != nil {
+			r.setVIPConfiguredCondition(
+				metav1.ConditionFalse,
+				"NamespaceDeletionError",
```

Comment:
For VirtualIPPool object, yes we can but for ClusterConfig, I was thinking about only one "global" condition `Ready` and then the SubReconciler can manage the Conditions they want, but ... :thinking:  I can still do an enum, it just means that if a SubReconciler add a Condition they need to update the ENUM.

Ok fine, I will do it

Reply:
Not sure it is possible in Golang but can an Enum extend another one ?

Reply:
So I did not define a dedicated Type for it, since it was a mess then because we couldn't have used the `metav1.Condition` type directly (or we should have built a function to convert from one to the other). But I defined a `const` close to the CRD per conditions